src/assets/styles/_base.scss

```diff
@@ -83,6 +83,9 @@ ol {
     margin-top: $kui-space-50;
   }
 }
+p:empty {
+  display: none;
+}
```

Comment:
**Question**: Why was this added? We shouldn’t have empty paragraphs anywhere. If we do, we’re doing something wrong there.

As an aside, this will only work for `<p></p>` but not `<p> </p>` (i.e. paragraphs with a text node).

Reply:
This is why I made our i18n service allow empty strings `''` instead of `' '`.

Now, if you don't set a i18n string for a paragraph, we don't display the paragraph. Previously we had JS conditionals checking to see if the string was set or not to more or less do the same thing. I can't think of an occasion where you'd want a completely empty `p` tag to be displayed.

Reply:
We should definitely use `v-if` directives to avoid rendering empty paragraphs instead.

> I can't think of an occasion where you'd want a completely empty p tag to be displayed.

Yes, so let’s not render them.

Reply:
We aren't rendering them, it has `display: none`. They literally don't have any display. It also means we reduce the amount of javascript code/conditional forking code we have to write as this little one-liner works for the entire project from The Outside. Set and forget.

Reply:
As long as they are in the DOM, they are rendered. They might not be painted, but rendered they are.

Sure, it would be nice to not need to write conditionals for optional text, but this comes at the cost of leaving empty DOM nodes around. That can come with problems in CSS where, for example, hidden elements participate in flex and grid contexts as items (and so they influence layout).

Reply:
I'm pretty sure whilst that might be the case for non-p tags, for p tags specifically I don't think this is an issue. I also think the cases where we actually specify `''` for a i18n string are incredibly slim.

If it does cause an issue with CSS flex there are other alternatives remove them, worst case, just add the `v-if` for that case. To be honest, we probably shouldn't be using flex layouts with `p` tags anyway, I can't think of a place where we already do or would want to.

As for the "cost" of of leaving the odd (probably single digit across the entire application) HTML tag I don't think that's an issue.

Lastly, this gives people editing i18n strings the opportunity to remove a piece of text (and paragraph) rather than leave a screen-rendered/displayed empty tag that possibly has padding/margins etc.

All in all the benefits of this outweigh any cost.